### PR TITLE
logger

### LINKED ISSUE #684: Write log-output to buffer
For some tests we need to check the error-output or other from the log. Some methods already exist to put this into a buffer. This issue asks to make those methods public.

### PR DESCRIPTION
**Pull request by ineiti**
Closes #684

### CODE REVIEW

File: log/lvl_test.go [hunk 1]
```diff
@@ -15,16 +15,16 @@ func init() {
 
 func TestTime(t *testing.T) {
 	SetDebugVisible(1)
-	getStdOut()
+	GetStdOut()
```
Comment (nikkolasg) [inline log/lvl_test.go]: ? Why is this line required/useful ?

File: log/log.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	stdOut = os.Stdout
+	stdErr = os.Stderr
+}
+
+var bufStdOut bytes.Buffer
+var wrStdOut *bufio.Writer
+var bufStdErr bytes.Buffer
+var wrStdErr *bufio.Writer
+
+func StdToBuf() {
```
Comment (nikkolasg) [inline log/log.go]: Mhhh API wise, if we want the log to go to another place than STDOUT, then it might make more sense to give the opportunity to the user to give the buffer it self ? For example, you might wanna log it to an external file or even to a connection.

At the same time, since the log package has no "instantiation" required (i.e. no `log = NewLog()`), having that in a thread safe manner might be a hassle...

File: log/log.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	wrStdOut = bufio.NewWriter(&bufStdOut)
+	stdOut = wrStdOut
+	wrStdErr = bufio.NewWriter(&bufStdErr)
+	stdErr = wrStdErr
```
Comment (nikkolasg) [inline log/log.go]: I don't understand why you need both a `Writer` and a `bytes.Buffer` since the latter implements the former ? I think you don't have to create a wrapper around, although it's friday night and I'm not sure :D

File: log/log.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+func StdToOs() {
+	stdOut = os.Stdout
+	stdErr = os.Stderr
+}
+
+func GetStdOut() string {
```
Comment (nikkolasg) [inline log/log.go]: * `Get` or no `Get` ... I tend for the latter, you ? (i don't really care, just wanna know if it's a decision or just an unintentional act)  
* StdOut is for *standard* output, but here it can also be the buffer output, so ... `OutputString()` & `ErrorString()` ? I don't know, these are not great names but I'm not a fan of `Stdout` for "output stream" and `stderr` for "error stream".

What do you think ?

File: log/log.go [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+	wrStdOut.Flush()
+	ret := bufStdOut.String()
+	bufStdOut.Reset()
+	return ret
+}
+
+func GetStdErr() string {
```
Comment (nikkolasg) [inline log/log.go]: Moreover, a comment should be there, also explaining that it flushes the "queue" so calling twice `GetXXX` won't result in the same output.

File: log/log.go (hunk 2 above)
Comment (ineiti) [inline log/log.go]: I'll say that's issue #689 ;)

File: log/lvl_test.go (hunk 1 above)
Comment (ineiti) [inline log/lvl_test.go]: this tests the output of the logger, so we need to flush it with a `Get`.

File: log/log.go (hunk 4 above)
Comment (ineiti) [inline log/log.go]: `stdout` and `stderr` are age-old unix names - so I don't dare touching those ;)

I like the `Get` here. It's shorter than `GetAndFlush`, and for me it's better than just `StdOut`...

Comment (nikkolasg) [inline log/log.go]: Ok thanks, you made me read the wikipedia on stdout and stderr and I finally got it. My confusion was that I thought we refer to *stdout* only for the *standard* output stream (i.e. the default/normal/common assigned output, by the terminal emulator or whatever). In fact *stdout* refers to **any** of the output stream the process uses.
So sorry, and I agree here then :)
